### ISSUE TITLE
Singleton Init and `MPI_Session_get_num_psets()`

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#10749]
### Configuration

```
./configure --prefix=/home/devel/mpi/openmpi/dev --without-ofi --without-ucx --with-pmix=internal --with-prrte=internal --with-libevent=internal --with-hwloc=internal --enable-debug --enable-mem-debug --disable-man-pages --disable-sphinx
```

### Reproducer

* Python code:
```python
from mpi4py import MPI

session = MPI.Session.Init()
num = session.Get_num_psets()
session.Finalize()
```

* Execution (singleton init, without mpiexec):
```
python script.py
```

### Behavior
Deadlock at the `num = session.Get_num_psets()` line.

**Comment by dalcinl** [comment]
@hppritcha You may want to take a look at this one.

**Comment by hppritcha** [comment]
closed via #10885 and #10891
